crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -296,9 +295,22 @@ impl<TYPES: NodeType> TestableNetworkingImplementation<TYPES>
                         .unwrap()
                 };
                 let bootstrap_addrs_ref = Arc::clone(&bootstrap_addrs);
+                let node_ids_ref = Arc::clone(&node_ids);
                 let da = da_keys.clone();
                 let reliability_config_dup = reliability_config.clone();
                 Box::pin(async move {
+                    // If it's the second time we are starting this network, clear the bootstrap info
+                    let mut write_ids = node_ids_ref.write().await;
+                    let mut clear = false;
+                    if write_ids.contains(&node_id) {
+                        write_ids.clear();
+                        clear = true;
+                    }
+                    write_ids.insert(node_id);
+                    drop(write_ids);
+                    if clear {
+                        bootstrap_addrs_ref.write().await.clear();
+                    }
```

Comment:
Can we just move this `clear` call to line 306? Then we don't need the `clear` boolean.

Reply:
Yeah I think so, was trying to rule out deadlocking by holding both the locks at once, but it wasn't the issue